kcqrs-example/src/main/kotlin/com/clouway/kcqrs/example/adapter/http/ProductRestApi.kt

```diff
@@ -0,0 +1,66 @@
+package com.clouway.kcqrs.example.adapter.http
+
+import com.clouway.kcqrs.core.AggregateNotFoundException
+import com.clouway.kcqrs.core.MessageBus
+import com.clouway.kcqrs.core.Repository
+import com.clouway.kcqrs.example.commands.RegisterProductCommand
+import com.clouway.kcqrs.example.domain.Product
+import spark.Request
+import spark.Response
+import spark.Route
+import java.util.*
+import javax.servlet.http.HttpServletResponse
+
+/**
+ * @author Miroslav Genov (miroslav.genov@clouway.com)
+ */
+class RegisterProductHandler(private val messageBus: MessageBus) : Route {
+    override fun handle(request: Request, response: Response): Any {
+        val uuid = UUID.randomUUID()
+
+        val req = request.readJson<RegisterProductRequestDto>(RegisterProductRequestDto::class.java)
+        messageBus.send(RegisterProductCommand(uuid, req.name))
+
+        return ProductAddedResponseDto(uuid.toString(), req.name)
+    }
+
+}
+
+
+class GetProductHandler(private val eventRepository: Repository) : Route {
+    override fun handle(request: Request, response: Response): Any? {
+        val id = request.params("id")
+
+        return try {
+            val product = eventRepository.getById(UUID.fromString(id), Product::class.java)
+            ProductDto(product.name)
+        } catch (ex: AggregateNotFoundException) {
+            response.status(HttpServletResponse.SC_NOT_FOUND)
+            ""
+        }
+    }
+
+}
+
+class ChangeProductNameHandler(private val eventRepository: Repository) : Route {
+    override fun handle(request: Request, response: Response): Any? {
+        val id = request.params("id")
+
+        val req = request.readJson<ChangeProductNameRequest>(ChangeProductNameRequest::class.java)
+
+        val product = eventRepository.getById(UUID.fromString(id), Product::class.java)
```

Comment:
this load doesn't look good to me, maybe because I expect 
val product = productRepository.get(UUID.fromString(id)) 
I thought that the productRepository will be the snapshot repository. 
Are we gone load the snapshot form the same microservice?(or we can use adapter from the datastore(snapshot) microservice)

Reply:
It depends. Each microservice will create it's own snapshots depending on the needs or this may be delegated to separate service about this. In all cases querying for Snapshot need to be performed either locally or through HTTP client which adds some additional overhead.